### ISSUE TITLE
RUF009: False positives in case of immutable objects

### ISSUE THREAD

**Issue by aberres** [charliermarsh/ruff#4091]
I'd argue the following code is ok, as all default parameters are immutable.

```python
import datetime as dt
from dataclasses import dataclass
from pathlib import Path

MyId = int

@dataclass
class Example:
    timedelta: dt.datetime = dt.timedelta(hours=1)

    date: dt.date = dt.date(2042, 1, 1)

    id: MyId = MyId(5)

    path: Path = Path()

e = Example()
```

`ruff` is not happy though:

```
ruff009.py:9:30: RUF009 Do not perform function call `dt.timedelta` in dataclass defaults
ruff009.py:11:21: RUF009 Do not perform function call `dt.date` in dataclass defaults
ruff009.py:13:16: RUF009 Do not perform function call `MyId` in dataclass defaults
ruff009.py:15:18: RUF009 Do not perform function call `Path` in dataclass defaults
```

Isn't there a similar case for default values of method parameters? I think this rule had a list of (at least some) allowed calls.

EDIT: Here is the list: https://github.com/charliermarsh/ruff/blob/main/crates/ruff/src/rules/flake8_bugbear/rules/function_call_argument_default.rs

**Comment by charliermarsh** [comment]
I agree.

**Comment by charliermarsh** [comment]
We should just add the same immutable-type checks as we do for the bugbear rules. \cc @mosauter if this is interesting to you.

**Comment by mosauter** [comment]
I created a PR for this. But have to say the type-alias `MyId` is AFAIK not something that ruff can detect (yet?). This would mean resolving the actual type behind a generic call. 

Which would in turn also enable us to detect a `NamedTuple` type which would technically also be a RUF009-False-Positive at this point in time.

Please correct me if I'm wrong on that.
